presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -1024,6 +1027,31 @@ public void testComplex()
     }
 
     @Test
+    public void testBucketedCatalog()
+            throws Exception
+    {
+        String bucketedCatalog = bucketedSession.getCatalog().get();
+        String bucketedSchema = bucketedSession.getSchema().get();
+
+        TableMetadata ordersTableMetadata = getTableMetadata(bucketedCatalog, bucketedSchema, "orders");
+        assertEquals(ordersTableMetadata.getMetadata().getProperties().get(BUCKETED_BY_PROPERTY), ImmutableList.of("custkey"));
+        assertEquals(ordersTableMetadata.getMetadata().getProperties().get(BUCKET_COUNT_PROPERTY), 11);
+
+        TableMetadata customerTableMetadata = getTableMetadata(bucketedCatalog, bucketedSchema, "customer");
+        assertEquals(customerTableMetadata.getMetadata().getProperties().get(BUCKETED_BY_PROPERTY), ImmutableList.of("custkey"));
+        assertEquals(customerTableMetadata.getMetadata().getProperties().get(BUCKET_COUNT_PROPERTY), 11);
+    }
+
+    @Test
+    public void testBucketedExecution()
+            throws Exception
+    {
+        assertQuery(bucketedSession, "select count(*) a from orders t1 join orders t2 on t1.custkey=t2.custkey");
+        assertQuery(bucketedSession, "select count(*) a from orders t1 join customer t2 on t1.custkey=t2.custkey", "SELECT count(*) from orders");
+        assertQuery(bucketedSession, "select count(distinct custkey) from orders");
```

Comment:
I'd add a `select count() from customer where custkey = 42` == 1.  That will verify that bucket filtering is working also (unless there is already a test for that).